### ISSUE TITLE
iOS tap delay

### ISSUE THREAD

**Issue by mnordine** [bp74/StageXL#218]
I get the dreaded iOS tap delay, even when trying 3rd party libraries, such as [FastClick](https://github.com/ftlabs/fastclick).

I created a [simple web app with stagehand](https://gist.github.com/mnordine/b9bad3dc898c03bc9601)

**Comment by nilsdoehring** [comment]
At least in that example, using MouseEvent.MOUSE_DOWN (or its touch equivalent) should do the trick. 

And in more complex situations: animation on DOWN, 'submit' on UP. If you need dragging: start a listener to ENTER_FRAME after a certain delay on DOWN, and cancel the delay and listener on UP; also only 'submit' if the mousePos didn't change (apply a certain threshold).

**Comment by mnordine** [comment]
But isn't that exactly what the library should be doing behind the scenes?

**Comment by bp74** [comment]
Hi guys. I don't know how or if the FastClick library is working, but the general advice is to register touch-events to get an immediate response on mobile devices. Please check out the

    var isSupported = StageXL.environment.isTouchEventSupported;
    StageXL.stageOptions.inputEventMode = InputEventMode.MouseOnly; 

You can use the same event handlers for mouse and touch events, since all MouseEvents and TouchEvents extend the same base class called InputEvent. This should make it pretty easy to get optimal support for mouse and touch.

**Comment by mnordine** [comment]
I tried to set it to InputEventMode.MouseOnly, but same result, still have the delay.

**Comment by mnordine** [comment]
Seems InputEventMode.MouseOnly is the default anyway.

**Comment by bp74** [comment]
yes this was just a copy-paste from an example, if you want to use touch input you have to set it to one of the touch modes. So you have to set the InputEventMode and also register to the touch events, this should give you instant feedback on user input.

**Comment by bp74** [comment]
You can also look at the example here:
https://github.com/bp74/StageXL_Samples/blob/master/example/interactive/drag/index.dart

**Comment by mnordine** [comment]
So, you get instant feedback on iOS if you set to `InputEventMode.TouchOnly` and use `onTouchBegin`. But I don't think this is a good solution, since every developer will have to implement something like @nilsdoehring suggested above for acceptable click behavior on iOS.

It'd be nice if devs could use `MouseEvent.CLICK`, not have iOS lag, and have it abstracted away for them, so they needn't worry about it.

**Comment by bp74** [comment]
Well we could do this, but the solution with a common event handler is not so much harder to implement. You could even use a helper function that adds both event handlers at the same time.

**Comment by nilsdoehring** [comment]
That's why I am in the process of creating the stagexl-commons library,
which aims to solve, well, common stuff like this :-)
It's not on pub yet, so you need to reference this github repo:
https://github.com/blockforest/stagexl-commons/

For your use case, extend the Button class and override downAction and/or
upAction.
https://github.com/blockforest/stagexl-commons/blob/master/lib/src/view/interact/button/impl/button.dart

Disclaimer: stagexl-commons is not an official part of StageXL. Also, no
animals have been harmed during its production.
